### ISSUE TITLE
List of sharable env vars

### ISSUE THREAD

**Issue by mal** [github/hubot#875]
Many plugins use the same information, the various twitter plugins are a prime example. Some agree on ENV var names, but others don't. It'd be good to have some sort of managed/recommended list of shareable env vars and exactly what users should populate them with/plugin writers can expect to be in them.

Weather plugins are another example that spring to mind:

```
HUBOT_LATITUDE - latitude in decimal degrees
HUBOT_LONGITUDE - longitude in decimal degrees
HUBOT_ELEVATION - ???
```

In the meantime maybe we could start the list here with suggestions?

**Comment by technicalpickles** [comment]
Good idea :+1: 

This might be hard to do up front, especially as scripts start to drain out of the main hubot-scripts repo. It'd basically take people running into specific combinations to know they are using similar things.

If there's enough use of it, I could imagine some small library (not a script) that these scripts could share to use the same environment variable, and have common error handling.

> In the meantime maybe we could start the list here with suggestions?

Basically any scripts that use the same HTTP APIs. Things like twitter, google, jira, github come to mind.

I'd also throw out some examples like temperature unit (since you mentioned weather), zip code, address.

**Comment by mal** [comment]
Maybe `hubot` core could help out a bit too; something like `robot.env` which given the env vars:

```
HUBOT__LATITUDE=10.152348
HUBOT_FORECAST_LATITUDE=14.954737
```

Would populate an object for different plugins such that most would see:

```json
{"latitude": 10.152348}
```

But `hubot-forecast` would see:

```json
{"latitude": 14.954737}
```

Then you get all the benefits of shared env vars, plus the ability to override them for specific plugins via very guessable alternatives, with `HUBOT__` being the global space and `HUBOT_{uppercase plugin name as on npm}_` as the specific plugin space. This would avoid conflicts purely by dint of the `npm` registry not conflicting. This approach wouldn't break anything already written, but would provide a nicer unified way going forward.
